Fix failure to find libmikmod:

checking for libmikmod - version >= 3.1.5... no

Because of:

error: implicitly declaring library function 'strlen' with type 'unsigned long (const char *)' [-Werror,-Wimplicit-function-declaration]
error: implicitly declaring library function 'strcpy' with type 'char *(char *, const char *)' [-Werror,-Wimplicit-function-declaration]
--- configure.orig	2011-07-31 19:31:11.000000000 -0500
+++ configure	2024-01-10 02:23:19.000000000 -0600
@@ -12947,6 +12947,7 @@
 #include <mikmod.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 char* my_strdup (char *str)
 {
@@ -12961,7 +12962,7 @@
   return new_str;
 }
 
-int main()
+int main(void)
 {
   int major,minor,micro;
   int libmikmod_major_version,libmikmod_minor_version,libmikmod_micro_version;
